enterprise/api/v1/views.py

```diff
@@ -213,6 +213,110 @@ def course_enrollments(self, request, pk):
 
         return Response(serializer.errors, status=HTTP_400_BAD_REQUEST)
 
+    @action(detail=True, methods=['post'], permission_classes=[permissions.IsAuthenticated])
+    @permission_required('enterprise.can_enroll_learners', fn=lambda request, pk: pk)
+    # pylint: disable=invalid-name,unused-argument
+    def enroll_learners_in_courses(self, request, pk):
+        """
+        Creates a set of licensed enterprise_learners by bulk enrolling them in the specified course.
+
+        Note: This endpoint isn't transactional in that if an enrollment fails, there can potentially be a scenario
+        with a subset of users enrolled. Requested learners' eligibility to enroll are expected to be pre-vetted.
+
+        Expected params:
```

Comment:
Why have you chosen this endpoint structure? Is there a really good reason not have this properly flattened into a list of desired enrollments of the form: [{email: A, course_key: B, license_uuid: C}, {email: D, course_key: E, license_uuid: F}] ?  This design choice is hard to read and requires dictionary traversal to do something that should be done in a loop.

Reply:
You highlight exactly what the small voice in my head was quietly nagging about. The parameter structure for the endpoint I think became unwieldy when I realized we needed more information. Allow me to do a refactor and we can assess a flattened structure.

Reply:
we would also have to include the course mode as a key in those sets of dictionaries.

Reply:
That's fine- over time we may find we actually have to add other properties as well, which gets un-readable unless the structure is very flat.

Reply:
I like the simplification made by making the params a simple array of things.In fact, that 'thing' can probably use a good name! Should we call it something like `licensed_enrollment_request`? Having a name can really help us talk about what specifically we are speaking of. 

Naming rationale: It's a request for enrollment, it may or may not be fulfilled. It has a license. It may have other fields but that is fine.

Reply:
as of the time writing this comment, we're using `licenses_info`